internal/provider/vdcg/dynamic_security_group_resource.go

```diff
@@ -0,0 +1,391 @@
+package vdcg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+
+	supertypes "github.com/FrangipaneTeam/terraform-plugin-framework-supertypes"
+
+	"github.com/orange-cloudavenue/cloudavenue-sdk-go/pkg/urn"
+	v1 "github.com/orange-cloudavenue/cloudavenue-sdk-go/v1"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/client"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/metrics"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/provider/common/mutex"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &DynamicSecurityGroupResource{}
+	_ resource.ResourceWithConfigure   = &DynamicSecurityGroupResource{}
+	_ resource.ResourceWithImportState = &DynamicSecurityGroupResource{}
+)
+
+// NewDynamicSecurityGroupResource is a helper function to simplify the provider implementation.
+func NewDynamicSecurityGroupResource() resource.Resource {
+	return &DynamicSecurityGroupResource{}
+}
+
+// DynamicSecurityGroupResource is the resource implementation.
+type DynamicSecurityGroupResource struct {
+	client   *client.CloudAvenue
+	vdcGroup *v1.VDCGroup
+}
+
+// Init Initializes the resource.
+func (r *DynamicSecurityGroupResource) Init(ctx context.Context, rm *DynamicSecurityGroupModel) (diags diag.Diagnostics) {
+	var err error
+
+	idOrName := rm.VDCGroupName.Get()
+	if rm.VDCGroupID.IsKnown() && urn.IsVDCGroup(rm.VDCGroupID.Get()) {
```

Comment:
If vdcGroupID is know then it's a vdcgroup, i think you don't need to check with func isvdcgroup